runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -96,43 +96,27 @@ uint8_t *J9::ARM64::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterat
    TR::Compilation* comp = TR::comp();
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(_cg->fe());
    TR_SharedCache *sharedCache = fej9->sharedCache();
-   TR::SymbolValidationManager *symValManager = comp->getSymbolValidationManager();
-
-   TR_VirtualGuard *guard;
-   uint8_t flags = 0;
-   TR_ResolvedMethod *resolvedMethod;
-
-   uint8_t *cursor = relocation->getRelocationData();
-
    TR_RelocationRuntime *reloRuntime = comp->reloRuntime();
    TR_RelocationTarget *reloTarget = reloRuntime->reloTarget();
 
-   uint8_t * aotMethodCodeStart = (uint8_t *) comp->getRelocatableMethodCodeStart();
-   // size of relocation goes first in all types
-   *(uint16_t *) cursor = relocation->getSizeOfRelocationData();
+   uint8_t flags = 0;
 
-   cursor += 2;
+   uint8_t *cursor         = relocation->getRelocationData();
+   uint8_t targetKind      = relocation->getTargetKind();
+   uint16_t sizeOfReloData = relocation->getSizeOfRelocationData();
 
-   uint8_t modifier = 0;
-   uint8_t *relativeBitCursor = cursor;
-   TR::LabelSymbol *table;
-   uint8_t *codeLocation;
+   // Zero-initialize header
+   memset(cursor, 0, sizeOfReloData);
```

Comment:
This is extremely necessary. Prior to this change, the flags was written in via a pointer. Now, it is written in using the `setFlag` API, which **masks it in**. Therefore, we need to make sure that buffer does not contain random data or it could result in that random data being interpreted as valid bits in the flag field.